polaris-service/src/main/java/org/apache/polaris/service/context/RealmContextResolver.java

```diff
@@ -25,11 +25,21 @@
 import org.apache.polaris.service.config.HasEntityManagerFactory;
 
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
-public interface RealmContextResolver extends Discoverable, HasEntityManagerFactory {
-  RealmContext resolveRealmContext(
+public abstract class RealmContextResolver implements Discoverable, HasEntityManagerFactory {
```

Comment:
By using composition in the class `DefaultContextResolver`, we can still support multiple implementations of `RealmContextResolver` by still keeping it as an interface, and providing an impl. with detailed logic.

Reply:
Technically we could still leave it as an interface and keep the inheritance relationship, but it would mean we have to make the implementations implement the default realm logic which seems unnecessary.

Reply:
+1 to keep it as an interface. Please ignore my composition idea.